homeassistant/components/random/binary_sensor.py

```diff
@@ -33,20 +35,32 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Random binary sensor."""
-    name = config.get(CONF_NAME)
-    device_class = config.get(CONF_DEVICE_CLASS)
 
-    async_add_entities([RandomSensor(name, device_class)], True)
+    async_add_entities([RandomBinarySensor(config)], True)
 
 
-class RandomSensor(BinarySensorEntity):
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Initialize config entry."""
+    async_add_entities(
+        [RandomBinarySensor(dict(config_entry.options), config_entry.entry_id)], True
+    )
+
+
+class RandomBinarySensor(BinarySensorEntity):
     """Representation of a Random binary sensor."""
 
-    def __init__(self, name, device_class):
+    _state: bool | None = None
+
+    def __init__(self, config: dict[str, Any], entry_id: str | None = None) -> None:
```

Comment:
```suggestion
    def __init__(self, config: Mapping[str, Any], entry_id: str | None = None) -> None:
```

So we don't need to convert the MappingProxingType to a dict